src/assets/TokensController.test.ts

```diff
@@ -35,19 +48,22 @@ describe('TokensController', () => {
   });
 
   it('should add token', async () => {
+    supportsInterfaceStub = sinon.stub().returns(Promise.resolve(false));
```

Comment:
What is this trying to do :thinking: 

It seems like you're trying to make the `supportsInterface` return `false` when it's called in this test, but I don't see how this could work. `supportsInterfaceStub` was `undefined` when `beforeEach` was run, and reassigning the variable here won't change what was passed into the `callsFake` call. I suspect this is only affecting the _next_ test, not this one.

But that doesn't make sense either - it seems to work later on :thinking: I am confused.

Either way, shared mocks like this should not be used in unit tests. It introduces a subtle dependency between tests, and it can make them very difficult to understand.

Reply:
Oh I think I understand. The `callsFake` function isn't executed until later, after it has been reassigned. 

Still seems bad for tests to depend upon code run in previous tests though.

Reply:
Yeah makes sense. Was trying to save some duplicated code but I will add the full stub to each test that requires it.